Src/KBCsv/Internal/CsvParser_Async.cs

```diff
@@ -159,7 +163,13 @@ internal sealed partial class CsvParser
                                 buffer[i] = this.values.GetDataRecordAndClear(headerRecord, this.valueBuilder.GetValueAndClear());
                                 break;
                             }
-                            else if (ch == Constants.LF)
+                            else if (this.lineTerminatorOverride == null && ch == Constants.LF)
```

Comment:
Couldn't these two cases be collapsed into one? That is, `if (ch == Constants.LF || ch == this.lineTerminatorOverride)`